### ISSUE TITLE
imxeglgvivsink: egl_platform_wayland

### ISSUE THREAD

**Issue by AstroJimX** [Freescale/gstreamer-imx#256]
Any guidelines on how to extend gst_imx_egl_viv_sink_egl_platform_init_window to use the window_handle (wl_surface) to display video on an application's window (GTK)? Currently, gst_imx_egl_viv_sink_egl_platform_init_window() gets passed window_handle but does not use it. As a result, video stream (GStream) using imxeglvivsink would create its own surface.

**Comment by AstroJimX** [comment]
In simpler terms, when building gstreamer-imx plugins under Wayland, the gst_imx_egl_viv_sink_egl_platform_init_window() function (see [https://github.com/Freescale/gstreamer-imx/blob/master/src/eglvivsink/egl_platform_wayland.c#L510]) gets the window_handle, but it does not use it. My thinking is that something similar to the waylandsink plugin from [https://github.com/GStreamer/gst-plugins-bad/tree/master/ext/wayland](gst-plugins-bad) is needed to pass the wl_display (display_handle) along with the wl_surface (window_handle). However, I am not sure how to integrate on top of the EGL window that is created by imxeglvivsink.

**Comment by dv1** [comment]
imxeglvivsink was written when glimagesink was not able to directly transfer frames to the GPU via DMA. Since then, glimagesink became able to handle DMA-BUF backed frames & frames whose memory is a GstPhysMemory instance, so that sink became obsolete. It is no longer present in gstreamer-imx. glimagesink, waylandsink, kmssink (with force-modesetting=true) are the alternatives (the latter on imx8 only).
